topi/python/topi/nn/softmax.py

```diff
@@ -20,10 +20,12 @@ def softmax(x, axis=-1):
     output : tvm.Tensor
         output shape is the same as input
     """
-    assert axis < len(x.shape)
+    if axis >= len(x.shape):
```

Comment:
move this after the axis transformation on negative values, so that things can work for axis that are negative